### ISSUE TITLE
Gitea backend for CMS

### ISSUE THREAD

**Issue by jimafisk** [plentico/plenti#271]
The people have spoken and they want Gitea backend support for the CMS: https://twitter.com/plentico/status/1615857323292987397

**Blocker:** Can't commit multiple files to a generic endpoint using the API: https://github.com/go-gitea/gitea/issues/14619

Per these [API Docs](https://crib.utwente.nl/git/api/swagger#/) there are only the 2 following methods for manipulating repo files:
- **PUT** `/repos​/{owner}​/{repo}​/contents​/{filepath}` Update a file in a repository ([link](https://crib.utwente.nl/git/api/swagger#/repository/repoUpdateFile))
- **POST** `​/repos​/{owner}​/{repo}​/contents​/{filepath}` Create a file in a repository ([link](https://crib.utwente.nl/git/api/swagger#/repository/repoCreateFile))

**Comment by jimafisk** [comment]
We've also been having trouble getting PKCE to work clientside: https://discourse.gitea.io/t/oauth-pkce-example/6573

Sounds like you can't omit the `client_secret` in Gitea currently: https://github.com/go-gitea/gitea/issues/17107

**Comment by jimafisk** [comment]
There's been progress on both the API endpoint and PKCE client secret: https://github.com/decaporg/decap-cms/issues/1109#issuecomment-1561721560 :tada: 

- Multi-file API endpoint (https://github.com/go-gitea/gitea/pull/24887) - Merged :heavy_check_mark: 
- PKCE optional client secret (https://github.com/go-gitea/gitea/pull/25033) - Open :eyes:

**Comment by jimafisk** [comment]
In testing this, I was getting the following error after being redirected back to my website: 

```
Only refresh_token or authorization_code grant type is supported
```

I forgot that you need to explicitly set the `Content-Type` when making your POST/fetch to request the access token (https://forum.gitea.com/t/oauth-pkce-example/6573):

```
headers: {
    "Content-Type": "application/x-www-form-urlencoded"
}
```

**Comment by jimafisk** [comment]
Gitea support should be available in [v0.7.16](https://github.com/plentico/plenti/releases/tag/v0.7.16) :tada: 

Just note we added a new `provider` key to the `cms` section in `plenti.json`:

```
"cms": {
	"provider": "gitea",
	"repo": "https://your-git-host.com/your-org/your-repo",
	"redirect_url": "https://your-static-site.com",
	"app_id": "your-oauth-id-from-gitea",
	"branch": "master"
}
```

The `provider` key is helpful since you can self-host Gitea and GitLab, which would make the repo live at a different URL than the officially supported ones (gitlab.com and gitea.com) which wouldn't necessarily communicate which platform you're on. If you omit this key, it defaults back to GitLab to ensure backwards compatibility with older version of Plenti. 

Note that since the APIs have not diverged too much at this point, this release also supports the Gitea fork [Forgejo](https://forgejo.org/) and you can use that as a provider name if you prefer.
